python/gtsam/examples/Pose2ISAM2Example.py

```diff
@@ -0,0 +1,155 @@
+    # update calls are required to perform the relinearization.
+    parameters = gtsam.ISAM2Params()
+    parameters.setRelinearizeThreshold(0.1)
+    parameters.setRelinearizeSkip(1)
+    isam = gtsam.ISAM2(parameters)
+
+    # The ground truth odometry measurements (without noise) of the robot during the trajectory.
+    odom_arr = [(2, 0, 0),
+                (2, 0, math.pi/2),
+                (2, 0, math.pi/2),
+                (2, 0, math.pi/2),
+                (2, 0, math.pi/2)]
+
+    # The initial estimates for robot poses 2-5. Pose 1 is initialized by the prior.
+    # To demonstrate iSAM2 incremental optimization, poor initializations were intentionally inserted.
+    pose_est = [gtsam.Pose2(2.3, 0.1, -0.2),
+                gtsam.Pose2(4.1, 0.1, math.pi/2),
+                gtsam.Pose2(4.0, 2.0, math.pi),
+                gtsam.Pose2(2.1, 2.1, -math.pi/2),
+                gtsam.Pose2(1.9, -0.2, 0.2)]
+
+    graph.push_back(gtsam.PriorFactorPose2(1, gtsam.Pose2(0, 0, 0), PRIOR_NOISE))
+    initial_estimate.insert(1, gtsam.Pose2(0.5, 0.0, 0.2))
+
+    def determine_loop_closure(odom: np.ndarray, current_estimate: gtsam.Values,
+        xy_tol=0.6, theta_tol=0.3) -> int:
+        """Simple brute force approach which iterates through previous states
+        and checks for loop closure.
+
+        Args:
+            odom: Vector representing noisy odometry (x, y, theta) measurement in the body frame.
+            current_estimate: The current estimates computed by iSAM2.
+            xy_tol: Optional argument for the x-y measurement tolerance.
+            theta_tol: Optional argument for the theta measurement tolerance.
+        Returns:
+            k: The key of the state which is helping add the loop closure constraint.
+                If loop closure is not found, then None is returned.
+        """
+        if current_estimate:
+            prev_est = current_estimate.atPose2(i+1)
+            rotated_odom = prev_est.rotation().matrix() @ odom[:2]
+            curr_xy = np.array([prev_est.x() + rotated_odom[0],
+                                prev_est.y() + rotated_odom[1]])
+            curr_theta = prev_est.theta() + odom[2]
+            for k in range(1, i+1):
+                pose_xy = np.array([current_estimate.atPose2(k).x(),
+                                    current_estimate.atPose2(k).y()])
+                pose_theta = current_estimate.atPose2(k).theta()
+                if (abs(pose_xy - curr_xy) <= xy_tol).all() and \
+                    (abs(pose_theta - curr_theta) <= theta_tol):
+                        return k
+
+    current_estimate = None
+    for i in range(len(odom_arr)):
+        # The "ground truth" change between poses
+        odom_x, odom_y, odom_theta = odom_arr[i]
+        # Odometry measurement that is received by the robot and corrupted by gaussian noise
+        odom = mult_gauss(odom_arr[i], ODOMETRY_NOISE.covariance())
```

Comment:
I would create all noisy measurements outside the loop, where previously you created `pose_est`